DMCompiler/DM/DMExpression.cs

```diff
@@ -39,7 +39,8 @@ public enum IdentifierPushResult {
 
         public static void Emit(DMObject dmObject, DMProc proc, DMASTExpression expression, DreamPath? inferredPath = null) {
             var expr = Create(dmObject, proc, expression, inferredPath);
-            expr.EmitPushValue(dmObject, proc);
+            // expr can be null if we error elsewhere
+            expr?.EmitPushValue(dmObject, proc);
```

Comment:
Wouldn't no `EmitPushValue` call happening here due to a null `expr` lead to ill-formed bytecode?